### DISCUSSION TITLE
Release of 0.54.8 broke semantic versioning

### DISCUSSION THREAD

**Discussion by alansouzati** [stylus/stylus#2608] [General]
Thanks for this package, we use at a few internal apps at Netflix.

Recently one of our apps (a pretty old one) just broke out of the blue. After a few hours investigating this, it seems like this [commit](https://github.com/stylus/stylus/commit/a1102a53e3fca4d531d9f962e8687c1f618a1e32) is the culprit.

The problem with this update is that [mkdirp 1.X](https://github.com/isaacs/node-mkdirp/issues/10) requires Node 10. Although it is definitely a good idea to stay up to date with dependencies, this completely breaks semantic versioning.

Our npm build used to work before, and now it is breaking because this, without even touching our package json.

Could you please release 0.54.9 that reverts this upgrade and then release stylus 1.X that does the proper semantic version updates with breaking changes?

**Comment by picklechips** [comment]
We're encountering the same issue. +1 to the suggestion.

As a workaround had to include a `npm-shrinkwrap.json` file to override the version of stylus in our dependency.

**Comment by AshotN** [comment]
Semantic versioning states 0.x.x is to be considered unstable and is not subject to the same rules. 

> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

**Comment by alansouzati** [comment]
Fair point @AshotN . Although unstable releases are meant to be temporary. It seems this library has been in that state for quite a while. And I do see this as a very common pattern for other libraries. I don’t think folks are doing that intentionally but they don’t fully understand the versioning structure. All that aside, using 0.X makes it really difficult for library consumers that want to use ^ and ~

**Comment by AshotN** [comment]
I agree, Stylus definitely needs a 1.0 release.

**Comment by piyushchauhan2011** [comment]
Was wondering any updates on this .

**Comment by iChenLei** [comment]
I think the biggest issue is lack stable CI infra, CI not passed when xdan update `mkdirp` deps.
![2021-08-21 22 51 19](https://user-images.githubusercontent.com/14012511/130325700-0fad59ab-0543-4f8c-bf8e-4d027e4853af.png)

but not add breaking change log in release note ([History.md](https://github.com/stylus/stylus/blob/dev/History.md)).
Stylus test self in node v10/v12/v14/16 via github actions CI now. ([ci.yml](https://github.com/stylus/stylus/blob/dev/.github/workflows/ci.yml))
